### PR TITLE
Creates a Rockon for the IDS NetAlertX

### LINKED ISSUE #396: NetAlertX rockon
i have made a netalertx rockon. creating a PR now...

### PR DESCRIPTION
**Pull request by kanecko**
> To ease and accelerate the PR submission, please use the template below to provide all requested information.
> You can find guidelines on which docker image to use in the [Rockstor's documentation](http://rockstor.com/docs/contribute_rockons.html#which-docker-image-should-i-use), 
> as well as examples of proper formatting in other rock-ons ([here](https://github.com/rockstor/rockon-registry/blob/master/teamspeak3.json)
> and [here](https://github.com/rockstor/rockon-registry/blob/master/nextcloud-official.json))

Fixes #396  .

### General information on project
This pull request proposes to add a new rock-on for the following project:
- name: NetAlertX
- website: https://netalertx.com/
- description: It is a IDS and it uses net=host. It has a WebUI reminiscent of pi-hole.

### Information on docker image
- docker image: https://hub.docker.com/r/jokobsk/netalertx
- is an official docker image available for this project?: Yes.

### Checklist
- [x] Passes [JSONlint](https://jsonlint.com) validation
- [x] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [x] `"description"` object lists and links to the docker image used
- [x] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [x] `"website"` object links to project's main website

### CODE REVIEW

**Comment by kanecko** [comment]
I've installed this rockon locally and opened the WebUI page. It detected a few devices from the network which is nice.
I will fiddle around with it some more, and report back, if I see any deal-breakers.
NOTE: this is yet another rockon that would benefit from default/optional values. Maybe I should continue the project...

File: netalertx.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                                "environment": {
+                                        "PORT": {
+                                                "description": "Enter the WebUI Web-port number. Recommended: 20211. This port has to be the same as the Rockon UI Link port, configured earlier. This ensures that the admin pages for NetAlertX are accessible via Web Browser",
+                                                "label": "Web-Port",
+                                                "index": 1
+                                        }
+                                }
+                        }
+                },
+                "description": "NetAlertX is an intrusion detection system (IDS).<p>Easily visualize all your networks in one place, enhance security with real-time alerts, and enjoy seamless integration with your smart home—no matter what hardware you use.</p><p>Based on the official docker image: <a href='https://hub.docker.com/r/jokobsk/netalertx' target='_blank'>https://hub.docker.com/r/jokobsk/netalertx</a>, available for amd64, arm64, armv7 and armv6 architecture.</p>",
```
Comment (Hooverdan96) [inline netalertx.json:40]:
```suggestion
                "description": "NetAlertX is an intrusion detection system (IDS).<p>Easily visualize all your networks in one place, enhance security with real-time alerts, and enjoy seamless integration with your smart home — no matter what hardware you use.</p><p>Based on the official docker image: <a href='https://hub.docker.com/r/jokobsk/netalertx' target='_blank'>https://hub.docker.com/r/jokobsk/netalertx</a>, available for amd64 and arm64 architecture.</p>",
```
Just to be consistent with Rockstor's focus on supporting those two architectures.

File: netalertx.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                                                "--net",
+                                                "host"
+                                        ]
+                                ],
+                                "launch_order": 1,
+                                "ports": {
+                                    "20211": {
+                                        "description": "Rockon UI Link Port. Recommended: 20211. This port has to be the same as the subsequent Web-port later in the configuration, so the admin pages can be accessed from within the Rockon page.",
+                                        "host_default": 20211,
+                                        "label": "Rockon_Web-Port",
```
Comment (Hooverdan96) [inline netalertx.json:17]:
```suggestion
                                        "label": "Rockon Web Port",
```

File: netalertx.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                                        },
+                                        "/app/db": {
+                                                "description": "Choose a share for NetAlertX database. E.g.: create a Share called netalertx-db for this purpose alone.",
+                                                "label": "NetAlertX db"
+                                        }
+                                },
+                                "environment": {
+                                        "PORT": {
+                                                "description": "Enter the WebUI Web-port number. Recommended: 20211. This port has to be the same as the Rockon UI Link port, configured earlier. This ensures that the admin pages for NetAlertX are accessible via Web Browser",
+                                                "label": "Web-Port",
```
Comment (Hooverdan96) [inline netalertx.json:34]:
```suggestion
                                                "label": "NetAlertX Web Port",
```

**Review by Hooverdan96 — COMMENTED** [review]
A couple of things I noted/suggested.
- some consistency around the labels
- reduced the architectures down to the 2 main ones Rockstor supports these days and be consistent how we updated the labels on the other Rockons.

**Comment by Hooverdan96** [comment]
@kanecko This is a great addition. thank you! I did a test install on a VM on my network:

image

image

As you can see in my review comments I updated a couple of the labels

image

image

image

image


Started up from within the Rockon page - after a while it started collecting a few devices on the network:

image

Fancy logs page:
image


Again, seems like nice addition to the repository, as it further enhances security layers and provides good visibility.

> NOTE: this is yet another rockon that would benefit from default/optional values. Maybe I should continue the project...

Please do, if you have the time. Contributors are always welcome, and, in fact, highly desired, considering that the current core dev team is really small.

**Comment by phillxnet** [comment]
@kanecko This looks nice. Thanks for sharing your work here.
@Hooverdan96 Let me know when this is ready, your suggestions to date look to have been integrated. Not sure if this was a first pass or the final review ready for publishing.

**Comment by Hooverdan96** [comment]
@phillxnet, this can be merged, as the suggestions were only cosmetic, not affecting the functionality.

**Comment by phillxnet** [comment]
Just done a quick install test prior to merge and got: "Built on: 2024-12-11 | Version: v24.10.31 ". This all looks good.

**Comment by kanecko** [comment]
FYI: after running this docker on default settings, I get the following spammed in logs:

> Jan 10 10:40:05 kernel: device eth0 entered promiscuous mode
> Jan 10 10:40:06 kernel: device eth0 left promiscuous mode

Apparently it is due to how the app scans the network. Every 5 mins it is sending an arp which apparently needs the promiscuous mode.

**Pull request merged**